### ISSUE TITLE
Guidance for presenting dark theme on website/workspaces

### ISSUE THREAD

**Issue by mcoker** [patternfly/patternfly-design#1152]
tl;dr - We need a way to show dark theme stuff on the website and in our development workspaces. How should we do that?

As we build out dark theme support and integrate into products, the PF and products teams need to be able to reference dark theme things on the website (and in our core/react/org workspaces, which use the same theme as the website). I suppose there are 2 approaches here:

1. We add a way to toggle dark theme at the website level - ie, the website itself has a "dark mode" and that dark mode applies to all of the component examples, demos, etc., too.
  * The tricky part here would be that the website/workspaces have a lot of custom styles that do not automagically adapt when dark theme is applied (like the products we're currently updating to better work with dark theme). This means we would likely need to make a pass through the website and update a number of things to:
    * Use vars instead of static values - ie, use `--pf-global--Color-100` instead of `#151515`. This is a transparent change, no visible change on the website.
    * Address some custom styles - ie, default page sections (like the home page and examples pages) use a light grey as the background that is custom, instead of using our standard bg's `--pf-global--BackgroundColor--100` or `--200`. We should consider either using our existing global vars/colors there instead, introduce a new background color into PF and dark theme that adds that light grey to our library, or just create a custom dark theme override for that color in dark theme for the website.
    * Use PF components where we aren't currently - ie, use the sidebar component for the main content and jump links, which should address potential background color issues.
  * There is an issue for this work and a draft PR
    * Issue - https://github.com/patternfly/patternfly-org/issues/2887
    * PR - https://github.com/patternfly/patternfly-org/pull/2929
1. We add dark theme toggles per example/demo.
  * The good - this would leave the overall website the way it is (less work there, even though the issues in https://github.com/patternfly/patternfly-org/issues/2887 would still be good to address)
  * The bad - dark theme is currently setup to work at the page level, which means we couldn't just scope dark theme styles to an example box the way examples are applied to the website currently. We would likely need to rewrite the way the examples are contained so so that we could embed page level styles in an individual example (ie, place the example in an iframe, and apply dark theme to the iframe), or extend the ability for dark theme to apply to just a section of the page instead of the entire page. 

cc @mcarrano @mceledonia
